crates/holochain_state/src/scratch.rs

```diff
@@ -74,9 +74,14 @@ impl Scratch {
         }
     }
 
-    pub fn add_header(&mut self, item: SignedHeaderHashed, chain_top_ordering: ChainTopOrdering) {
+    pub fn add_header(
+        &mut self,
+        zome: Option<Zome>,
```

Comment:
In what situation does a header not have a zome?

Reply:
@freesig init

Reply:
This is a little at odds with how validation works as it treats Init as running for all zomes not none but I'm not sure that matters?